src/Nest/CommonAbstractions/SerializationBehavior/SerializerFactory.cs

```diff
@@ -0,0 +1,41 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Elasticsearch.Net;
+using Newtonsoft.Json;
+
+namespace Nest
+{
+	public interface ISerializerFactory
+	{
+		IElasticsearchSerializer Create(IConnectionSettingsValues settings);
+
+		IElasticsearchSerializer CreateStateful(IConnectionSettingsValues settings, JsonConverter converter);
```

Comment:
If they're only an internal implementation detail, I think we shouldn't expose a public method contract for creating them - maybe split this interface into two (SOL**I**D), `ISerializerFactory` and `IStatefulSerializerFactory`, the latter being a non-public interface?